### PR TITLE
Excel DataLink

### LINKED ISSUE #9281: Excel format support in datalinks
We want to be able to read Excel files within our Data Links.

This is not a separate data link as the Excel format can be in various places - on S3, in the Enso Cloud, on a webpage. Instead we need to extend our `Format` to support `Excel`.

We need to be able to:
- read the whole workbook, as a 'connection',
- read a particular sheet (as a Table),
- read a particular range (as a Table).

### PR DESCRIPTION
**Pull request by radeusgd**
### Pull Request Description

- Adds the Excel format as one of the formats supported when creating a data link.
	- The data link can choose to read the file as a workbook, or read a sheet or range from it as a table, like `Excel_Format`.
- Also updated Delimited format dialog to allow customizing the quote style.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by radeusgd** [comment]
Excel Data Link

image
image

**Comment by radeusgd** [comment]
Customizing quote style in Delimited

image
image
image

File: app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json [hunk 1]
```diff
@@ -220,6 +235,124 @@
...
+      "required": ["type", "subType", "section"]
+    },
+
+    "ExcelSectionWorkbook": {
+      "title": "Workbook",
+      "type": "object",
+      "properties": {
+        "type": { "const": "workbook", "type": "string" },
+        "defaultSheet": {
+          "title": "Default sheet",
```
Comment (jdunkerley) [inline app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json:263]: `New Sheet Name`? 
We could change on both sides.

File: distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Format.enso [hunk 2]
```diff
@@ -169,14 +170,17 @@ Delimited_Format.from (that : JS_Object) =
...
         . if_not_nothing (Encoding.from_name encoding_name)
         . if_nothing Encoding.utf_8
-    headers = that.get "headers" . if_nothing Infer
+    headers = that.get "headers" |> parse_boolean_with_infer "headers"
     skip_rows = that.get "skip_rows" . if_nothing 0
     row_limit = that.get "row_limit"
     keep_invalid_rows = that.get "keep_invalid_rows" . if_nothing True
+    quote_style = case that.get "quote_style" of
+        Nothing -> Quote_Style.With_Quotes
+        json -> Quote_Style.from json
```
Comment (jdunkerley) [inline distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Format.enso:179]:
```suggestion
    quote_style = if that.get "quote_style" . is_nothing then Quote_Style.With_Quotes else Quote_Style.from json
```

**Review by jdunkerley — APPROVED** [review]

File: app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json (hunk 1 above)
Comment (radeusgd) [reply app/ide-desktop/lib/dashboard/src/data/dataLinkSchema.json:263]: I don't think it's clearer.

`Default sheet (for writing)` would be most clear IMO, as it says that it is the _default name_ used _when writing to the workbook_.

Without knowing the context "New sheet name" doesn't tell me much more than "default", other than it _is_ a bit clearer that this is writing-only.

Maybe we should add a `ℹ️` icon at the end, to hint that the user should hover over this to see the `description` which I think provides enough context?

File: distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Format.enso (hunk 2 above)
Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Format.enso:179]: This can't work - `json` is undefined.

Comment (radeusgd) [reply distribution/lib/Standard/Table/0.0.0-dev/src/Delimited/Delimited_Format.enso:179]: It'd have to be
```ruby
quote_style = if that.get "quote_style" . is_nothing then Quote_Style.With_Quotes else Quote_Style.from (that.get "quote_style")
```
but then we are duplicating `that.get "quote_style"` which I think looks bad. The whole purpose of the `case of` was to write `that.get "quote_style"` once but avoid giving it a separate name (I don't like temporary variables like `quote_style_json` next to the actual `quote_style` and case of allows to hide that)

**Review by GregoryTravis — APPROVED** [review]

**Pull request merged**